lib/fluent/plugin/filter_kubernetes_metadata.rb

```diff
@@ -272,11 +290,11 @@ def get_metadata_for_record(match_data, container_id, create_time, batch_miss_ca
         'pod_name'        => pod_name
       }
       if @kubernetes_url.present?
-        pod_metadata = get_pod_metadata(container_id, namespace_name, pod_name, create_time, batch_miss_cache)
+        pod_metadata = get_pod_metadata(cache_key, namespace_name, pod_name, create_time, batch_miss_cache)
 
-        if (pod_metadata.include? 'containers') && (pod_metadata['containers'].include? container_id)
-          metadata['container_image'] = pod_metadata['containers'][container_id]['image']
-          metadata['container_image_id'] = pod_metadata['containers'][container_id]['image_id']
+        if (pod_metadata.include? 'containers') && (pod_metadata['containers'].include? container_name)
```

Comment:
I have concerns changing to `container_name` from `container_id` will cause caching issues, as in maybe uniqueness is no longer guaranteed.

Reply:
In this case, there is no pod metadata and thus no container metadata anyways?